S09-hashes/objecthash.t

```diff
@@ -140,6 +140,11 @@ plan 55;
     eval-lives-ok 'my %*a{Int}', "Accept dynamic object hash"
 }
 
+# https://github.com/rakudo/rakudo/issues/1486
+{
+    ok {:foo, (if 0.5.rand < 1 { :bar })}, 'a conditional inside circumfix {} works';
```

Comment:
I'd like to ask about a comment explaining the approach about `if`. It's going to be really confusing what purpose of it without the background of this PR.

BTW, a sufficiently sophisticated static optimizer would optimize away this `if` too. But, unfortunately, we're not going get one of the kind any time soon.